compiler/src/dotty/tools/dotc/decompiler/DecompilationPrinter.scala

```diff
@@ -0,0 +1,31 @@
+package dotty.tools.dotc
+package decompiler
+
+import dotty.tools.dotc.core.Contexts._
+import dotty.tools.dotc.core.Phases.Phase
+
+/** Phase that prints the trees in all loaded compilation units.
+ *
+ *  @author Nicolas Stucki
+ */
+class DecompilationPrinter extends Phase {
+
+  override def phaseName: String = "decompilationPrinter"
+
+  override def run(implicit ctx: Context): Unit = {
+    val unit = ctx.compilationUnit
+
+    val pageWidth = ctx.settings.pageWidth.value
+
+    val doubleLine = "=" * pageWidth
+    val line = "-" * pageWidth
+
+    println(doubleLine)
+    println(unit.source)
+    println(line)
+
+    val code = unit.tpdTree.show
+    println(if (ctx.useColors) printing.SyntaxHighlighting(code) else code)
```

Comment:
It should really be `printing.SyntaxHighlighting(code).mkString`. We shouldn't assume that the underlying `Iterable[Char].toString` works as we expect.

Reply:
I will remove the syntax highlighting as it is not compatible with all the output formats of `show`. I will work on coloring in a separate PR.